src/interfaces/node.h

```diff
@@ -110,6 +111,11 @@ class Node
     //! Disconnect node by id.
     virtual bool disconnectById(NodeId id) = 0;
 
+#ifdef ENABLE_EXTERNAL_SIGNER
```

Comment:
Would be nice to drop `#ifdef ENABLE_EXTERNAL_SIGNER` in the interface here and just return an empty vector in the implementation or throw a not-implemented exception. Cross-process interfaces are trying not to be conditional to avoid build complexity and fragile incompatibilities between builds.

Also maybe in general in this PR, preprocessor is a little more widely used that it needs to be. Preprocessor is great when you need to write platform specific fragments or cut out a significant external or internal dependency like boost or the wallet. But if you just want to act conditionally, normal code is a good way to go. I'd try to avoid conditional compilation except for the parts of the code directly depending on boost.

Reply:
I added a commit to reduce `#ifdef ENABLE_EXTERNAL_SIGNER` usage, including in the interface.